pkg/cmd/roachtest/tests/failure_injection.go

```diff
@@ -331,33 +349,247 @@ var latencyTest = func(c cluster.Cluster) failureSmokeTest {
+		if err != nil {
+			return readWriteBytesOverTime{}, err
+		}
+		// Drop the page caches on each VM to force them to read from disk. Without
+		// this, we don't run the workload long enough for the process to read
+		// from disk instead of memory.
+		c.Run(ctx, option.WithNodes(c.CRDBNodes()), "echo 3 | sudo tee /proc/sys/vm/drop_caches")
+		select {
+		case <-ctx.Done():
+			return readWriteBytesOverTime{}, ctx.Err()
+		case <-time.After(30 * time.Second):
+		}
+
+		afterStallReadBytes, afterStallWriteBytes, err := f.GetReadWriteBytes(ctx, l, c.Node(stalledNode).InstallNodes())
+		if err != nil {
+			return readWriteBytesOverTime{}, err
+		}
+		afterUnaffectedReadBytes, afterUnaffectedWriteBytes, err := f.GetReadWriteBytes(ctx, l, c.Node(unaffectedNode).InstallNodes())
+		if err != nil {
+			return readWriteBytesOverTime{}, err
+		}
+
+		return readWriteBytesOverTime{
+			stalledNodeReadBytes:     afterStallReadBytes - beforeStallReadBytes,
+			stalledNodeWriteBytes:    afterStallWriteBytes - beforeStallWriteBytes,
+			unaffectedNodeReadBytes:  afterUnaffectedReadBytes - beforeUnaffectedReadBytes,
+			unaffectedNodeWriteBytes: afterUnaffectedWriteBytes - beforeUnaffectedWriteBytes,
+		}, nil
+	}
+
+	var tests []failureSmokeTest
+	for _, stallWrites := range []bool{true, false} {
+		for _, stallReads := range []bool{true, false} {
+			if !stallWrites && !stallReads {
+				continue
+			}
+
+			nodes := c.CRDBNodes()
+			rand.Shuffle(len(nodes), func(i, j int) {
+				nodes[i], nodes[j] = nodes[j], nodes[i]
+			})
+			stalledNode := nodes[0]
+			unaffectedNode := nodes[1]
+
+			testName := fmt.Sprintf("%s: WritesStalled=%t ReadsStalled=%t", failures.CgroupsDiskStallName, stallWrites, stallReads)
+			tests = append(tests, failureSmokeTest{
+				testName:    testName,
+				failureName: failures.CgroupsDiskStallName,
+				args: failures.DiskStallArgs{
+					StallWrites:  stallWrites,
+					StallReads:   stallReads,
+					RestartNodes: true,
+					Nodes:        install.Nodes{install.Node(stalledNode)},
+				},
+				validateFailure: func(ctx context.Context, l *logger.Logger, c cluster.Cluster, f failures.FailureMode) error {
+					res, err := getReadWriteBytesOverTime(ctx, l, c, f.(*failures.CGroupDiskStaller), stalledNode, unaffectedNode)
+					if err != nil {
+						return err
+					}
+					l.Printf("ReadBytes and WriteBytes over last 30 seconds:%+v", res)
+
+					// We expect the stalled node to have essentially 0 bytes read/written to disk while
+					// stalled while the unstalled node should have been unaffected. It would be nice to
+					// assert that it is 0 (it usually is), but because of how cgroups throttles io,
+					// (throughput is not a hard limit and limits can accumulate for bursts of io) we
+					// sometimes see some throughput.
+					if stallWrites {
+						if res.unaffectedNodeWriteBytes < res.stalledNodeWriteBytes*10 {
```

Comment:
Is 10x over 30 seconds a safe threshold? I'm thinking it's workload-specific... lemme circle back to it, after having digested the whole PR.